test-it/karma/javascriptinvoke/Async.Test.js

```diff
@@ -297,41 +297,37 @@ describe('A JavaScript function invoke', function () {
                 const {rawPrint, rawPrintError} = await runSlicesAsync();
                 expect(rawPrint).toBeEmptyString();
                 expect(rawPrintError).toBeEmptyString();
-                expect(viPathParser('error.status')).toBeFalse();
-                expect(viPathParser('error.code')).toBe(0);
-                expect(viPathParser('error.source')).toBeEmptyString();
-                expect(viPathParser('return')).toBe(49);
-                expect(error.message).toMatch(/retrieved more than once/);
+                expect(viPathParser('error.status')).toBeTrue();
+                expect(viPathParser('error.code')).toBe(44300);
+                expect(viPathParser('error.source')).toMatch(/retrieved more than once/);
+                expect(viPathParser('return')).toBe(0);
+                expect(error.message).toMatch(/threw an error, so this callback cannot be invoked/);
             };
         });
         it('using the completion callback', async function () {
             window.SingleFunction = function (input, jsapi) {
                 var completionCallback = jsapi.getCompletionCallback();
-                try {
-                    jsapi.getCompletionCallback();
-                } catch (ex) {
-                    error = ex;
-                }
+                jsapi.getCompletionCallback();
                 // TODO mraj this allows completing after an exception is thrown, this should not be allowed.
                 // Instead if we are not resolved we should resolve immediately with an error
                 completionCallback(input * input);
             };
-            await test();
+            test().catch(function (ex) {
+                error = ex;
+            });
         });
         it('using promises', async function () {
             window.SingleFunction = async function (input, jsapi) {
                 // By returning a promise the first implicit call happens
                 await delayForTask();
-                try {
-                    jsapi.getCompletionCallback();
-                } catch (ex) {
-                    error = ex;
-                }
+                jsapi.getCompletionCallback();
                 // TODO mraj this allows completing after an exception is thrown, this should not be allowed.
                 // Instead if we are not resolved we should resolve immediately with an error
                 return input * input;
             };
-            await test();
+            test().catch(function (ex) {
```

Comment:
cleaner because it avoids mixing async and promise patterns

Reply:
That's what we were discussing earlier. Try/catch doesn't actually catch the exception being thrown because of the Promise code in module_javaScriptInvoke.js, which already has a .catch, but this does seem to work.

Reply:
> Try/catch doesn't actually catch the exception being thrown

That's right I forgot about that

> but this does seem to work

Which part works? Using the expect shouldn't work for the Promise case for the same reason...

Reply:
Actually, it does work for the promise case. When I added the .catch to the test call like it is here, the error variable gets set correctly.